src/app/model.ts

```diff
@@ -1,36 +1,127 @@
-import {Tickets, TicketType, Priority} from 'node-zendesk';
+import {Tickets} from 'node-zendesk';
 
-import {configStore} from '../store';
+import {AppFormValue, AppFormValues} from 'mattermost-redux/types/apps';
 
-type CreateTicketFormValues = {
-    subject: string;
-    type: TicketType;
-    priority: Priority;
-    additional_message: string;
-    post_message: string;
-}
+import {AppFieldNames, ZDFieldValidation, getMultiselectValues} from '../utils';
+
+import {configStore} from '../store';
 
-export const getTicketFromForm = (values: CreateTicketFormValues): Tickets.CreatePayload => {
-    const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+export type FieldValidationErrors = {[name: string]: string};
 
-    const additionalMessage = values.additional_message || '';
-    const postMessage = values.post_message || '';
+interface ITicketFromFrom {
+    getTicket(): Tickets.CreatePayload;
+    fieldValidationErrors: FieldValidationErrors;
+}
 
-    const zdMessage = additionalMessage + '\n' +
-            postMessage + '\n' +
-            mmSignature;
+export class TicketFromForm implements ITicketFromFrom {
+    formValues: AppFormValues;
+    ticket: Tickets.CreateModel;
+    fieldValidationErrors: FieldValidationErrors
 
-    const ticket: Tickets.CreatePayload = {
-        ticket: {
-            subject: values.subject,
-            type: values.type,
-            priority: values.priority,
+    constructor(values: AppFormValues) {
+        this.formValues = values;
+        this.fieldValidationErrors = {};
+        this.ticket = {
             comment: {
-                body: zdMessage,
+                body: this.getPostMessage(),
             },
-        },
-    };
-    return ticket;
-};
+        };
+    }
+
+    getPostMessage(): string {
+        const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+
+        const additionalMessage = this.formValues[AppFieldNames.AdditionalMessage] || '';
+        const postMessage = this.formValues[AppFieldNames.PostMessage] || '';
+
+        const zdMessage = additionalMessage + '\n' +
+                postMessage + '\n' +
+                mmSignature;
+
+        return zdMessage;
+    }
+
+    getTicket(): Tickets.CreatePayload {
+        this.mapFormValuesToTicket();
+        return {ticket: this.ticket};
+    }
+
+    // mapFormValuesToTicket returns a zendesk ticket with AppCall values mapped
+    // 1:1 to a zendesk ticket
+    mapFormValuesToTicket(): void {
```

Comment:
This mapping prob. could be generalized in the future for any third-party tickets/forms.

Reply:
@catalin.  I cleaned this code up a bit. Please see https://github.com/mattermost/mattermost-app-zendesk/commit/4eeb93eb2376365f528f86ae12d65e12e1c4d827

mapFormValuesToTicket() now operates in the following manner:

* get pruned form values from `pruneFormFields()`
* iterate on only those fields that are returned 
* now each pruned field will be mapped 1:1 with a zendesk ticket field
* only two functions remain in the foreach loop:
  * `mapCustomFieldToTicket()`
  * `mapFieldToTicket()`

Hopefully this makes the intention cleaner 

Also introduced two small helper functions
`isCustomField()`
`isOmittedField()`